src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -316,8 +319,198 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+
+        var guidMatches = guidPattern.Matches(expression);
+        var dateTimeMatches = dateTimePattern.Matches(expression);
+        var localDateTimeMatches = localDateTimePattern.Matches(expression);
+        var randomIntMatches = randomIntPattern.Matches(expression);
+        var timestampMatches = timestampPattern.Matches(expression);
+
+        if (guidMatches.Count > 0)
+        {
+            expressionResolution = Guid.NewGuid().ToString();
+            return true;
+        } else if(dateTimeMatches.Count > 0)
+        {
+            expressionResolution = ParseDateTime(expression, dateTimeMatches, dateTimePattern);
+            return true;
+        } else if(localDateTimeMatches.Count > 0)
+        {
+            expressionResolution = ParseDateTime(expression, localDateTimeMatches, localDateTimePattern);
+            return true;
+        } else if(randomIntMatches.Count > 0)
+        {
+            expressionResolution = ParseRandInt(expression, randomIntMatches);
+            return true;
+        }else if(timestampMatches.Count > 0)
+        {
+            expressionResolution = ParseTimestamp(expression, timestampMatches);
+            return true;
+        }
+        return false;
+    }
+
+    protected string ParseTimestamp(string expression, MatchCollection matches)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+        for (int i = matches.Count - 1; i >= 0; i--)
+        {
+            DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+            Match match = matches[i];
+            if (match.Groups.Count == 3)
+            {
+                try
+                {
+                    if (match.Groups["offset"].Success && match.Groups["option"].Success
+                        && int.TryParse(match.Groups["offset"].Value, out int offset)
+                        && offset != 0)
+                    {
+                        dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                    }
+
+                    text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToUnixTimeSeconds().ToString(), text.Substring(match.Index + match.Value.Length));
+                }
+                catch (Exception)
+                {
+                    // If we can't parse the offset, just leave the variable in place
```

Comment:
Consider delegating the binding failure result creation down to these functions. For example, this function could return `node.CreateBindingSuccess(text);` in the success case and `node.CreateBindingFailure(HttpDiagnostics.SomeDiagnostic(ex.Message));` in the failure case.